inspect_image_test.go

```diff
@@ -199,42 +199,181 @@ func testInspectImage(t *testing.T, when spec.G, it spec.S) {
+				when("Platform API < 0.4", func() {
+					when("CNB_PROCESS_TYPE is set", func() {
+						it.Before(func() {
+							h.AssertNil(t, fakeImage.SetEnv("CNB_PROCESS_TYPE", "other-process"))
+						})
+
+						it("returns processes setting the correct default process", func() {
+							info, err := subject.InspectImage("some/image", useDaemon)
+							h.AssertNil(t, err)
+
+							h.AssertEq(t, info.Processes,
+								ProcessDetails{
+									DefaultProcess: &launch.Process{
+										Type:    "other-process",
+										Command: "/other/process",
+										Args:    []string{"opt", "1"},
+										Direct:  true,
+									},
+									OtherProcesses: []launch.Process{
+										{
+											Type:    "web",
+											Command: "/start/web-process",
+											Args:    []string{"-p", "1234"},
+											Direct:  false,
+										},
+									},
+								},
+							)
+						})
+					})
+
+					when("CNB_PROCESS_TYPE is set, but doesn't match an existing process", func() {
+						it.Before(func() {
+							h.AssertNil(t, fakeImage.SetEnv("CNB_PROCESS_TYPE", "missing-process"))
+						})
+
+						it("returns a nil default process", func() {
+							info, err := subject.InspectImage("some/image", useDaemon)
+							h.AssertNil(t, err)
+
+							h.AssertEq(t, info.Processes,
+								ProcessDetails{
+									DefaultProcess: nil,
+									OtherProcesses: []launch.Process{
+										{
+											Type:    "other-process",
+											Command: "/other/process",
+											Args:    []string{"opt", "1"},
+											Direct:  true,
+										},
+										{
+											Type:    "web",
+											Command: "/start/web-process",
+											Args:    []string{"-p", "1234"},
+											Direct:  false,
+										},
+									},
+								},
+							)
+						})
 					})
 
-					it("returns processes setting the correct default process", func() {
+					it("returns a nil default process when CNB_PROCESS_TYPE is not set and there is no web process", func() {
+						h.AssertNil(t, fakeImage.SetLabel(
+							"io.buildpacks.build.metadata",
+							`{
+  "processes": [
+    {
+      "type": "other-process",
+      "command": "/other/process",
+      "args": ["opt", "1"],
+      "direct": true
+    }
+  ]
+}`,
+						))
+
 						info, err := subject.InspectImage("some/image", useDaemon)
 						h.AssertNil(t, err)
 
 						h.AssertEq(t, info.Processes,
 							ProcessDetails{
-								DefaultProcess: &launch.Process{
-									Type:    "other-process",
-									Command: "/other/process",
-									Args:    []string{"opt", "1"},
-									Direct:  true,
-								},
+								DefaultProcess: nil,
 								OtherProcesses: []launch.Process{
 									{
+										Type:    "other-process",
+										Command: "/other/process",
+										Args:    []string{"opt", "1"},
+										Direct:  true,
+									},
+								},
+							},
+						)
+					})
+				})
+
+				when("Platform API >= 0.4", func() {
+					when("CNB_PROCESS_TYPE is set", func() {
```

Comment:
I don't think we expect this variable to be set in the image if platform api >= 0.4, but I suppose there is value in checking the edge cases.

Reply:
I only check it once, now, just to verify that we ignore it and follow entrypoint.